cryptographic_estimators/MQEstimator/mq_algorithm.py

```diff
@@ -56,7 +56,7 @@ def __init__(self, problem: MQProblem, **kwargs):
         if q is not None and not is_prime_power(q):
             raise ValueError("q must be a prime power")
 
-        if w is not None and not 2 <= w <= 3:
+        if  w < 2 or 3 < w:
```

Comment:
the `or` should be an `and`, right?

Reply:
and 3 and w should be swapped?

Reply:
forget what I wrote: I overlooked the `not` in the previous line. Sorry

Reply:
It is correct with an or. 
not 2 <= w <= 3 equals   w < 2 or 3 < w